Automattic/vip-go-mu-plugins
# Show notice if a non-prod site may be indexable

Creates an admin notice indicating the site may be indexable if `blog_public` option is `1`

![Screen Shot 2020-05-15 at 2 59 44 PM](https://user-images.githubusercontent.com/251183/82099668-c0bf4500-96bc-11ea-98ac-aa3a70768c31.png)

Additionally, does not show a notice if:
 - the current user does not have `manage_options`
 - the notice has been dismissed (saved in options, reset if blog_public is resaved to `1`)
 - the home url ends in `.go-vip.co` or `.go-vip.net`

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

Example:

1. Check out PR and/or otherwise run this file as an mu-plugin. Only dependency is `wp_endswith()` [from here](https://github.com/Automattic/vip-go-mu-plugins/blob/ef9d3ca1038275e7abe5f39fcbd12410506c4d04/vip-helpers/vip-utils.php#L348) which, despite the prefix, is in fact not a core function.
1. If needed, define `VIP_GO_APP_ENVIRONMENT` as anything but `production`
1. If you do not immediately see a notice in WP-Admin, blog_public may already be `0`, in Reading Options, toggle and resave the Search Engine Visibility  checkbox.
1. With blog_public set to `1`, the notice should be visible

Things to check:
1. clicking the x should dismiss the notice, and it _should not_ return on refresh
1. toggle and save the Search Engine Visibility setting, notice should be restored
1. change `VIP_GO_APP_ENVIRONMENT` to `production`, notice should be gone
1. check on a go-vip.co/net site, notice should not appear
